src/widgets/answers/answers.tsx

```diff
@@ -0,0 +1,168 @@
+import connectAnswers, {
+  AnswersRendererOptions,
+  AnswersConnectorParams,
+} from '../../connectors/answers/connectAnswers';
+
+const withUsage = createDocumentationMessageGenerator({ name: 'answers' });
+const suit = component('Answers');
+
+const renderer = ({
+  renderState,
+  cssClasses,
+  containerNode,
+  templates,
+}): Renderer<AnswersRendererOptions, Partial<AnswersWidgetParams>> => (
+  { hits, isLoading, instantSearchInstance },
+  isFirstRendering
+) => {
+  if (isFirstRendering) {
+    renderState.templateProps = prepareTemplateProps({
+      defaultTemplates,
+      templatesConfig: instantSearchInstance.templatesConfig,
+      templates,
+    });
+    return;
+  }
+
+  render(
+    <Answers
+      cssClasses={cssClasses}
+      hits={hits}
+      isLoading={isLoading}
+      templateProps={renderState.templateProps}
+    />,
+    containerNode
+  );
+};
+
+export type AnswersTemplates = {
+  /**
+   * Template to use for the header. This template will receive an object containing `hits` and `isLoading`.
+   */
+  header?: Template<{
+    hits: Hit[];
+    isLoading: boolean;
+  }>;
+
+  /**
+   * Template to use for the loader.
+   */
+  loader?: Template;
+
+  /**
+   * Template to use for each result. This template will receive an object containing a single record.
+   */
+  item?: Template<Hit>;
+};
+
+export type AnswersCSSClasses = {
+  /**
+   * CSS class to add to the root element of the widget.
+   */
+  root?: string | string[];
+
+  /**
+   * CSS class to add to the wrapping element when no results.
+   */
+  emptyRoot?: string | string[];
+
+  /**
+   * CSS classes to add to the header.
+   */
+  header?: string | string[];
+
+  /**
+   * CSS classes to add to the loader.
+   */
+  loader?: string | string[];
+
+  /**
+   * CSS class to add to the list of results.
+   */
+  list?: string | string[];
+
+  /**
+   * CSS class to add to each result.
+   */
+  item?: string | string[];
+};
+
+export type AnswersWidgetParams = {
+  /**
+   * CSS Selector or HTMLElement to insert the widget.
+   */
+  container: string | HTMLElement;
+
+  /**
+   * The templates to use for the widget.
+   */
+  templates?: AnswersTemplates;
+
+  /**
+   * The CSS classes to override.
+   */
+  cssClasses?: AnswersCSSClasses;
+};
+
+export type AnswersWidget = WidgetFactory<
+  AnswersRendererOptions,
+  AnswersConnectorParams,
+  AnswersWidgetParams
+>;
+
+const answersWidget: AnswersWidget = widgetParams => {
+  const {
+    container,
+    attributesForPrediction,
+    queryLanguages,
+    nbHits,
+    templates = defaultTemplates,
+    cssClasses: userCssClasses = {},
+  } = widgetParams || ({} as typeof widgetParams);
+
+  if (!container) {
+    throw new Error(withUsage('The `container` option is required.'));
+  }
+
+  const containerNode = getContainerNode(container);
+  const cssClasses = {
+    root: cx(suit(), userCssClasses.root),
+    emptyRoot: cx(suit({ modifierName: 'empty' }), userCssClasses.emptyRoot),
```

Comment:
why is it called emptyRoot when the generated class name is empty? Could you in parallel also open a PR on specs so we can discuss final DOM before it leaves experimental?

Reply:
What do you mean by
> when the generated class name is empty

![image](https://user-images.githubusercontent.com/499898/104345787-0445a980-54ff-11eb-96ea-912e182bd68c.png)

![image](https://user-images.githubusercontent.com/499898/104345840-10316b80-54ff-11eb-8aec-1b0ef8058c1a.png)

Reply:
maybe rootEmpty? what do we call this in the other widgets?

Reply:
We've got `emptyRoot`s in hits and infinite-hits.